ci_framework/roles/ci_network/tasks/main.yml

```diff
@@ -0,0 +1,68 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if environment file exists
+  register: env_file
+  ansible.builtin.stat:
+    path: "{{ cifmw_network_generated_layout }}"
+
+- name: Load generated network if exists
+  when:
+    - env_file.stat.exists
+  ansible.builtin.include_vars:
+    file: "{{  cifmw_network_generated_layout }}"
+
+- name: Chose appropriate data source
```

Comment:
Not really needed is this facts moves to https://github.com/openstack-k8s-operators/ci-framework/pull/586/files#diff-0f14100a05f379e84695ebb0a68b8a4b4b16bc145b0127b0450206dc9d941fbcR63 as a variables or maybe goes directory into the loop condition.

Reply:
right. I used it in the early iteration for debugging